cli/azd/pkg/config/config.go

```diff
@@ -91,46 +106,127 @@ func Load() (*Config, error) {
+	currentNode := c.data
+	parts := strings.Split(path, ".")
+	for _, part := range parts {
+		if depth == len(parts) {
+			currentNode[part] = value
+			return nil
+		}
+		var node map[string]any
+		value, ok := currentNode[part]
+		if !ok || value == nil {
+			node = map[string]any{}
+		}
+
+		if value != nil {
+			node, ok = value.(map[string]any)
+			if !ok {
+				return fmt.Errorf("failed converting node at path '%s' to map", part)
+			}
+		}
+
+		currentNode[part] = node
+		currentNode = node
+		depth++
 	}
 
-	return &config, nil
+	return nil
 }
 
-func getConfigFilePath() (string, error) {
-	configPath, err := GetUserConfigDir()
-	if err != nil {
-		return "", fmt.Errorf("failed locating config dir")
+// Removes any values stored at the specified path
+// When the path location is an object will remove the whole node
+// When the path does not exist, will return a `nil` value
+func (c *config) Unset(path string) error {
+	depth := 1
+	currentNode := c.data
+	parts := strings.Split(path, ".")
+	for _, part := range parts {
+		if depth == len(parts) {
+			delete(currentNode, part)
+			return nil
+		}
+		var node map[string]any
+		value, ok := currentNode[part]
+
+		// Path already doesn't exist, NOOP
+		if !ok || value == nil {
+			return nil
+		}
+
+		if value != nil {
+			node, ok = value.(map[string]any)
+			if !ok {
+				return fmt.Errorf("failed converting node at path '%s' to map", part)
+			}
+		}
+
+		currentNode[part] = node
+		currentNode = node
+		depth++
 	}
 
-	return filepath.Join(configPath, "config.json"), nil
+	return nil
 }
 
-type contextKey string
+// Gets the value stored at the specified location
+// Returns the value if exists, otherwise returns nil & a value indicating if the value existing
+func (c *config) Get(path string) (any, bool) {
+	depth := 1
+	currentNode := c.data
+	parts := strings.Split(path, ".")
+	for _, part := range parts {
+		if depth == len(parts) {
+			value, ok := currentNode[part]
+			return value, ok
+		}
+		value, ok := currentNode[part]
+		if !ok {
+			return value, ok
+		}
+
+		node, ok := value.(map[string]any)
+		if !ok {
+			return nil, false
+		}
+
+		currentNode = node
+		depth++
+	}
+
+	return nil, false
+}
+
+// Parses azd configuration JSON and returns a Config instance
+func Parse(configJson []byte) (Config, error) {
+	var data map[string]any
+	err := json.Unmarshal(configJson, &data)
+	if err != nil {
+		return nil, fmt.Errorf("failed unmarshalling configuration JSON: %w", err)
+	}
 
-const configContextKey contextKey = "config"
+	return NewConfig(data), nil
+}
 
 // Gets the AZD config from current context
 // If it does not exist will return a new empty AZD config
-func GetConfig(ctx context.Context) *Config {
-	config, ok := ctx.Value(configContextKey).(*Config)
-	if !ok {
-		loadedConfig, err := Load()
-		if err != nil {
-			loadedConfig = &Config{}
-		}
-		config = loadedConfig
+func GetConfig() Config {
+	azdConfig, err := Load()
+	if err != nil {
+		azdConfig = NewConfig(nil)
```

Comment:
This should throw an error if any unexpected errors occur. It should only create an empty config if the file doesn't exist, with potentially different handling for corrupted JSON.

Reply:
If we block on reading file error or invalid json unmarshalling then we get into a state where `azd` could become unusable till those items are resolved.

Yes, there could be some data loss.  Happy to discuss more.

Reply:
Updated to non-blocking when file does not exist, but will throw/show error on other error scenarios such as permissions issue or JSON unmarshelling.

Reply:
Updated error handling will be more explicit.  We handle case directly for when azd config file does not exist but will show the user if there are other types of errors like JSON parsing.